shapely/geometry/point.py

```diff
@@ -64,15 +64,19 @@ def __new__(self, *args):
             # Accept either (x, y) or [(x, y)]
             if not hasattr(coords, "__getitem__"):  # generators
                 coords = list(coords)
-
             coords = np.asarray(coords).squeeze()
         else:
             # 2 or 3 args
             coords = np.array(args).squeeze()
 
+        if coords.ndim > 1:
+            raise ValueError(
+                "Invalid values passed to Point constructor, "
+                "only size-1 arrays are accepted"
```

Comment:
You can remove this second part, as that is exactly the confusing error message from passing an array to the Decimal(..) constructor (which happens if ndim > 1) that we want to avoid raising by checking this manually here. 

Maybe we can append the first part with something like ".. can only pass 2 or 3 coordinates".

Reply:
`TypeError(f"Point() takes at most 3 arguments ({len(args)} given)")`  - is raised when more then 3 arguments are given
so maybe in a similar way: `ValueError(f"Point() takes only scalar or 1-size vector arguments, got {args}")`